drivers/sensor/lis2dh/lis2dh_trigger.c

```diff
@@ -185,13 +200,53 @@ static int lis2dh_start_trigger_int2(const struct device *dev)
 {
 	struct lis2dh_data *lis2dh = dev->data;
 	const struct lis2dh_config *cfg = dev->config;
+	int status = 0;
+	uint8_t reg = 0, mask = 0, val = 0;
 
 	setup_int2(dev, true);
 
-	return lis2dh->hw_tf->write_reg(
-		dev,
-		cfg->hw.anym_on_int1 ? LIS2DH_REG_INT1_CFG : LIS2DH_REG_INT2_CFG,
-		(cfg->hw.anym_mode << LIS2DH_INT_CFG_MODE_SHIFT) | LIS2DH_ANYM_CFG);
+	bool has_anyt = (lis2dh->handler_tap != NULL);
+	bool has_anym = (lis2dh->handler_anymotion != NULL);
```

Comment:
ok, so you are using the presence of an handler (set by SENSOR_TRIG_DELTA and/or SENSOR_TRIG_TAP) to start it. Good

Reply:
BTW, have you tested both TAP and ANY MOTION just to see if both works correctly? Please do... (thx!)

Reply:
yes, both were tested and are worked as expected